### ISSUE TITLE
write_csv truncating numbers?

### ISSUE THREAD

**Issue by teucer** [hadley/readr#117]
In the simple example below, there is a difference between df01 and df02:

```R
library(readr)

set.seed(1234)
n <- 1e3
dat <-  data.frame(a=as.integer(sample(1:1000,n,replace=TRUE)),
                 b=rnorm(n),
                 c=sample(c("foo","bar","baz","qux","quux"),n,replace=TRUE),
                 d=sample(c(TRUE,FALSE),n,replace=TRUE), 
                 stringsAsFactors=FALSE)

write.csv(dat,"test.csv",row.names = FALSE)
write_csv(dat,"test1.csv")

df01 <- read_csv("test.csv")
df02 <- read_csv("test1.csv")
```

**Comment by hadley** [comment]
A some what smaller reproducible example:

```R
input <- runif(100)
output <- read_csv(write_csv(data.frame(input), ""))$input

all.equal(input, output)
```

**Comment by hadley** [comment]
Ooops, have now gone too far in the opposite direction.

**Comment by hadley** [comment]
Closed 0e4cc5ec
